### ISSUE TITLE
Support weight decay for Adam optimizer

### ISSUE THREAD

**Issue by fabian-sp** [cafaxo/Llama2.jl#29]
Weight decay could be quite easily added to the Adam optimizer code (it would then become AdamW).

For example, before [this line](https://github.com/cafaxo/Llama2.jl/blob/77f7b8fa40bb5b96d7f44df16546aa576aeaef09/src/training/optimizers.jl#L116), add

`@turbo @. x *= 1-α*λ`

According to the LLama2 paper, training was done with value `λ=0.1`.
